posts/2021-08-30-binding-app-to-postgresql-db-with-odo-and-ol-stack.adoc

```diff
@@ -0,0 +1,284 @@
+
+[source,sh]
+----
+  databaseName: "sampledb"
+  databasePassword: "samplepwd"
+  databaseUser: "sampleuser"
+----
+
+Customize the Database resource name and add the needed annotations in `db.yaml`. The annotations are used to inform the Service Binding Operator about the service data required by the application.
+
+[source,sh]
+----
+metadata:
+  name: sampledatabase
+  annotations:
+    service.binding/db_name: 'path={.spec.databaseName}'
+    service.binding/db_password: 'path={.spec.databasePassword}'
+    service.binding/db_user: 'path={.spec.databaseUser}'
+----
+
+Generate the Database resource service devfile configuration.
+
+[source,sh]
+----
+odo service create --from-file db.yaml
+----
+
+Configuration added to devfile.yaml:
+
+[source,sh]
+----
+...
+- kubernetes:
+    inlined: |
+      apiVersion: postgresql.dev4devs.com/v1alpha1
+      kind: Database
+      metadata:
+        annotations:
+          service.binding/db_name: path={.spec.databaseName}
+          service.binding/db_password: path={.spec.databasePassword}
+          service.binding/db_user: path={.spec.databaseUser}
+        name: sampledatabase
+      spec:
+        databaseCpu: 30m
+        databaseCpuLimit: 60m
+        databaseMemoryLimit: 512Mi
+        databaseMemoryRequest: 128Mi
+        databaseName: sampledb
+        databaseNameKeyEnvVar: POSTGRESQL_DATABASE
+        databasePassword: samplepwd
+        databasePasswordKeyEnvVar: POSTGRESQL_PASSWORD
+        databaseStorageRequest: 1Gi
+        databaseUser: sampleuser
+        databaseUserKeyEnvVar: POSTGRESQL_USER
+        image: centos/postgresql-96-centos7
+        size: 1
+  name: sampledatabase
+...
+----
+
+Push the updates to the cluster.
+
+[source,sh]
+----
+odo push
+----
+
+So far, two things have taken place:
+
+- The application was deployed on your cluster. The application was built and deployed using Maven. However, you could have also built and deployed the application using Gradle by using the `odo create java-openliberty-gradle mysboproj` command instead when creating the Java Open Liberty component.
+
+- A Dev4Ddevs Database resource instance was created. This, in turn, triggered the creation of a PostgreSQL database instance.
+
+However, the application is still not usable because it does not have the data needed to connect to the database. Let's solve that next.
+
+=== Bind the application to the PostgreSQL service
+
+List the available services to which the application can be bound. The PostgreSQL database service should be listed.
+
+[source,sh]
+----
+odo service list
+----
+
+Output:
+
+[source,sh]
+----
+NAME                        MANAGED BY ODO      STATE      AGE
+...
+Database/sampledatabase     Yes (mysboproj)     Pushed     50s
+...
+----
+
+Generate the service binding devfile configuration.
+
+[source,sh]
+----
+odo link Database/sampledatabase
+----
+
+Configuration added to devfile.yaml:
```

Comment:
The following configuration is added to devfile.yaml:
--->
The following configuration is automatically added to `devfile.yaml`: